.azuredevops/deploy-platform-core-iam.yaml

```diff
@@ -0,0 +1,114 @@
+name: 1.0.$(BuildID)-${{ parameters.deployResources }}
+
+parameters:
+  - name: deployFromFeature
+    displayName: "Deploy from Feature Branch"
+    type: boolean
+    default: false
+  - name: deployResources
+    displayName: "Resources to Deploy"
+    type: string
+    default: "All"
+    values:
+      - All
+      - Cluster Custom Resources Reader Role
+pr:
+  branches:
+    include:
+      - main
+  paths:
+    include:
+      - infra/core/iam/*
+
+trigger:
+  batch: true
+  branches:
+    include:
+      - '*'
+  paths:
+    include:
+      - infra/core/iam/*
+
+resources:
+  repositories:
+    - repository: ADPPipelineCommon
+      name: DEFRA/adp-pipeline-common
+      endpoint: DEFRA
+      type: github
+      ref: main
+      
+variables:
+  - name: IsAll
+    value: ${{ eq(parameters.deployResources,'All') }}
+  - name: IsClusterCustomResourcesReaderRole
+    value: ${{ or(eq(variables.IsAll,true), eq(parameters.deployResources,'Cluster Custom Resources Reader Role')) }}    
+
+extends:
+  template: /pipelines/common-infra-deploy.yaml@ADPPipelineCommon
```

Comment:
can't this inherit ADP common? the abstraction layer for envs already exists?

Reply:
Removed env section.